src/main/java/Duke.java

```diff
@@ -37,342 +48,367 @@ public void run() throws DukeException {
             }
-            
-            void parse() throws DukeException {
-                try {
-                    if (nextLine.startsWith("done")) {
-                        inputs.taskDone(nextLine);
-                    } else if (nextLine.startsWith("remove")) {
-                        inputs.taskRemove(nextLine);
-                    } else if (nextLine.startsWith("todo")) {
-                        inputs.taskTodo(nextLine);
-                    } else if (nextLine.startsWith("deadline")) {
-                        inputs.taskDeadline(nextLine);
-                    } else if (nextLine.startsWith("event")) {
-                        inputs.taskEvent(nextLine);
-                    } else if (nextLine.equals("list")) {
-                        if (inputs.inputs.size() == 0) {
-                            System.out.println("No tasks in list");
-                        } else {
-                            System.out.println("Here are the tasks in your list:");
-                            int len = inputs.inputs.size();
-                            for (int i = 1; i <= len; i++) {
-                                Input inputType = inputs.inputs.get(i - 1);
-                                if (inputType.isDone) {
-                                    System.out.println(i + ". " + inputType.id + "[/] " + inputType.content +
-                                            inputType.printedTime); 
-                                } else {
-                                    System.out.println(i + ". " + inputType.id + "[x] " + inputType.content +
-                                            inputType.printedTime);
-                                    
-                void parse() throws DukeException {
-                    try {
-                        if (nextLine.startsWith("done")) {
-                            inputs.taskDone(nextLine);
-                        } else if (nextLine.startsWith("remove")) {
-                            inputs.taskRemove(nextLine);
-                        } else if (nextLine.startsWith("todo")) {
-                            inputs.taskTodo(nextLine);
-                        } else if (nextLine.startsWith("deadline")) {
-                            inputs.taskDeadline(nextLine);
-                        } else if (nextLine.startsWith("event")) {
-                            inputs.taskEvent(nextLine);
-                        } else if (nextLine.startsWith("find")) {
-                            inputs.taskFind(nextLine);
-                        } else if (nextLine.equals("list")) {
-                            if (inputs.inputs.size() == 0) {
-                                System.out.println("No tasks in list");
+            String keyword = nextLine.substring(5);
+            int len = inputs.size();
+            for(int i = 0; i < len; i++) {
+                Input input = inputs.get(i);
+                if (input.content.contains(keyword)) {
```

Comment:
Maybe you can consider making this more succint. In other words, try to avoid nesting conditionals which could get quite messy, though I am guilty of this too!

Reply:
Would the code more readable if you use extra method to handle the conditions? 🤔